### PR TITLE
feat(args): initial version of reporting multiple errors

### PR DESCRIPTION
**Pull request by i-keliukh**
This version only implements reporting multiple errors for the following parameters:
 - vcs type
 - config path
 - source path for local VCS

The idea is to commit this version and then gradually move the entire code base to this approach.

### CODE REVIEW

**Comment by i-keliukh** [comment]
Relates to #261

File: universum/modules/error_state.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from ..lib.gravity import Module, Dependency
+
+
+class GlobalErrorState(Module):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)  # type: ignore
+        self.errors: List[str] = []
+
+
+class ErrorState(Module):
```
Comment (o-andrieiev) [inline universum/modules/error_state.py:13]: I don't like the name of this class. If we keep it as-is, it implies that Launches 'is a' ErrorState, which is wrong.
I propose 'HasErrorState' or 'ErrorStorage', or even something like 'ErrorCognizant'

File: universum/modules/error_state.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def __init__(self, *args, **kwargs) -> None:
+        # mypy doesn't support cooperative inheritance
+        super().__init__(*args, **kwargs)  # type: ignore
+        self.global_error_state = self.global_error_state_factory()
+
+    def error(self, message: str):
+        self.global_error_state.errors.append(inspect.cleandoc(message))
+
+    def is_error_state(self):
```
Comment (o-andrieiev) [inline universum/modules/error_state.py:24]: did you mean 'is_in_error_state'?

File: universum/modules/error_state.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from typing import List
+import inspect
+
+from ..lib.gravity import Module, Dependency
+
+
+class GlobalErrorState(Module):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)  # type: ignore
```
Comment (o-andrieiev) [inline universum/modules/error_state.py:9]: codestyle question: should I use 'super()' in similar code, or 'super(GlobalErrorState, self)' ?

File: universum/modules/launcher.py [hunk 4]
```diff
@@ -289,7 +289,7 @@ def _handle_postponed_out(self):
 
 @needs_output
 @needs_structure
-class Launcher(ProjectDirectory):
+class Launcher(ProjectDirectory, ErrorState):
```
Comment (o-andrieiev) [inline universum/modules/launcher.py:292]: codestyle/convention question: should ErrorState be inherited first or last?

File: universum/modules/error_state.py (hunk 3 above)
Comment (i-keliukh) [reply universum/modules/error_state.py:9]: Use super().
super(GlobalErrorState, self) is only needed to support python 2.

File: universum/modules/error_state.py (hunk 2 above)
Comment (i-keliukh) [reply universum/modules/error_state.py:24]: Well, I think it is fine either way.
However, if 'is_in_error_state' seems better to you, I will fix it.

File: universum/modules/launcher.py (hunk 4 above)
Comment (i-keliukh) [reply universum/modules/launcher.py:292]: In python it only affects which members are shadowed.
If two bases are totally unrelated, the order doesn't matter.
I usually add a new base to the end of the list.
At least, this way I don't need to think about it too much.

File: universum/modules/error_state.py (hunk 1 above)
Comment (i-keliukh) [reply universum/modules/error_state.py:13]: Ok.

File: universum/modules/error_state.py (hunk 2 above)
Comment (o-andrieiev) [reply universum/modules/error_state.py:24]: I think 'is_in_error_state' is more clear

File: universum/modules/vcs/vcs.py [hunk 5]
```diff
@@ -142,7 +144,8 @@ def __init__(self, *args, **kwargs):
         self.api_support = self.api_support_factory()
 
         if self.settings.report_to_review:
-            self.code_review = self.driver.code_review()
+            if not self.is_error_state():
```
Comment (o-andrieiev) [inline universum/modules/vcs/vcs.py:147]: wrong method name

Comment (i-keliukh) [reply universum/modules/vcs/vcs.py:147]: Looks like PyCharm missed it

**Review by o-andrieiev — APPROVED** [review]

**Pull request merged**